lib/rpmsg/rpmsg_virtio.c

```diff
@@ -390,6 +390,8 @@ static void *rpmsg_virtio_get_tx_payload_buffer(struct rpmsg_device *rdev,
 		}
 	}
 
+	metal_assert(!wait || rp_hdr);
```

Comment:
This effectively changes the API for the function, since the previous behavior was just to return null when wait == 0 and getting the buffer failed. With this change, instead of an error return, the system aborts. This could break client code in a very bad way. I would prefer that this commit be reverted.

Reply:
You are right, I have reverted this commit.